File: BAEKJOON/java/10174_팰린드롬.java
import java.io.*;

public class Main {
    public static String s;
    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        int tc = Integer.parseInt(br.readLine());
        for (int i = 0; i < tc; i++) {
            input();
            solution();
        }

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        s = br.readLine();
    }

    private static void solution() throws IOException {
        s = s.toLowerCase();

        StringBuilder sb = new StringBuilder(s);
        String rev = sb.reverse().toString();

        if (rev.equals(s)) {
            bw.write("Yes\n");
            return;
        }

        bw.write("No\n");
    }
}

File: BAEKJOON/cpp/10174_팰린드롬.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <cctype>
using namespace std;

string s;

void input() {
	getline(cin, s);
}

void solution() {
	for (int i = 0; i < s.length(); i++) {
		s[i] = tolower(s[i]);
	}

	string rev(s);
	reverse(rev.begin(), rev.end());

	if (rev == s) {
		cout << "Yes\n";
		return;
	}

	cout << "No\n";
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	int tc;
	cin >> tc;
	cin.ignore();

	for (int i = 0; i < tc; i++) {
		input();
		solution();
	}

	return 0;
}